genie-web/src/main/java/com/netflix/genie/web/services/impl/LocalJobKillServiceImpl.java

```diff
@@ -176,6 +222,24 @@ public void onKillJobEvent(@NotNull final KillJobEvent event) throws GenieExcept
         this.killJob(event.getId(), event.getReason());
     }
 
+    @Override
+    public void registerCallbackForKill(
+        @NotBlank(message = "No id entered. Unable to register job for killing.") final String id,
+        @NotNull final Runnable callback
+    ) {
+
+        jobKillCallbacks.put(id, callback);
+
+    }
+
+    @Override
+    public void deregisterCallbackForKill(
+        @NotBlank(message = "No id entered. Unable to deregister job for killing.") final String id
+    ) {
+        jobKillCallbacks.remove(id);
+        log.info("Deregistered kill callback for job id {}", id);
```

Comment:
I think I meant you should check the return from the `remove` API to see if it was actually removed or not